### PR TITLE
[IR] Rename apis

### PR DESCRIPTION
**Pull request by justinchuby**
`def_node()` -> `producer()`
`def_index()` -> `index()`
`users()` -> `consumers()`

Fixes https://github.com/microsoft/onnxscript/issues/1379

### CODE REVIEW

File: onnxscript/ir/_core.py [hunk 1]
```diff
@@ -882,35 +882,59 @@ class OptionalType(_RecursiveTypeBase):
 
 
 class Value(_protocols.ValueProtocol, _display.PrettyPrintable):
-    """IR Value."""
+    """IR Value.
+
+    A value is a named entity that can be used to represent an input or output of a graph,
+    a function, or a node. The information it stores corresponds to ``ValueInfoProto``
```
Comment (gramalingam) [inline onnxscript/ir/_core.py:888]: Maybe more accurate to say that it generalizes `ValueInfoProto` ... it does store more info (producer, consumer, whether it is a constant, etc. etc.)

Comment (justinchuby) [inline onnxscript/ir/_core.py:888]:
```suggestion
    a function, or a node. The information it stores generalizes over ``ValueInfoProto``
```

File: onnxscript/ir/_core.py [hunk 2]
```diff
@@ -943,35 +967,35 @@ def __str__(self) -> str:
...
         return f"%{_quoted(value_name)}<{type_text},{shape_text}>"
 
-    def def_node(self) -> Node | None:
+    def producer(self) -> Node | None:
         """The node that produces this value."""
-        return self._def_node
+        return self._producer
 
-    def def_index(self) -> int | None:
+    def index(self) -> int | None:
```
Comment (gramalingam) [inline onnxscript/ir/_core.py:974]: nit: I wonder if `index` is too generic. But I can't think of a better alternative. `output_index` may not be that clear either. On second thought, seems ok to leave it as `index`

File: onnxscript/ir/_core.py (hunk 1 above)
Comment (justinchuby) [reply onnxscript/ir/_core.py]: Done

File: onnxscript/ir/_core.py [hunk 3]
```diff
@@ -943,35 +967,35 @@ def __str__(self) -> str:
...
+    def consumers(self) -> frozenset[tuple[Node, int]]:
+        """Return a set of consumers of the value.
 
         The set contains tuples of ``(Node, index)`` where the index is the index of the input
         of the node. For example, if ``node.inputs[1] == value``, then the user is ``(node, 1)``.
         """
-        return frozenset(self._users)
+        return frozenset(self._consumers)
 
     def _add_user(self, user: Node, index: int) -> None:
```
Comment (gramalingam) [inline onnxscript/ir/_core.py:986]: should this be renamed to `_add_consumer` ?

File: onnxscript/ir/_core.py [hunk 4]
```diff
@@ -943,35 +967,35 @@ def __str__(self) -> str:
...
         """Add a user node.
 
         This is an internal method. It should only be called by the Node class.
         """
-        self._users.add((user, index))
+        self._consumers.add((user, index))
 
     def _remove_user(self, user: Node, index: int) -> None:
```
Comment (gramalingam) [inline onnxscript/ir/_core.py:993]: Rename this too?

**Review by gramalingam — APPROVED** [review]

File: onnxscript/ir/_core.py (hunk 3 above)
Comment (justinchuby) [reply onnxscript/ir/_core.py]: Done

File: onnxscript/ir/_core.py (hunk 4 above)
Comment (justinchuby) [reply onnxscript/ir/_core.py]: Done

**Comment by justinchuby** [comment]
@titaiwangms I am going to merge this

**Pull request merged**